hoomd/hpmc/integrate.py

```diff
@@ -109,17 +107,18 @@ def __init__(self, seed, d, a, move_ratio, nselect, deterministic):
     def attach(self, simulation):
         '''initialize the reflected c++ class'''
         sys_def = simulation.state._cpp_sys_def
-        if simulation.device.mode == 'GPU':
+        if simulation.device.mode == 'gpu':
             self._cpp_cell = _hoomd.CellListGPU(sys_def)
             if simulation._system_communicator is not None:
                 self._cpp_cell.setCommunicator(simulation._system_communicator)
             self._cpp_obj = getattr(_hpmc,
-                                    self._cpp_cls + 'GPU')(sys_def,
```

Comment:
I appreciate the effort, but we currently don't have a 1-1 mapping between C++ exported class names and Python class names. I'm going to revert this to get tests pass again. Later we could bring this back as a simplification: #622